From 2e07500261accdee55e41d5d562bbda6968f4069 Mon Sep 17 00:00:00 2001
From: Herrie <Github.com@herrie.org>
Date: Mon, 13 Nov 2023 08:23:36 +0100
Subject: [PATCH] hybrisadaptor: Fix deprecated use of QMap insertMulti

insertMulti has been deprecated as per https://doc.qt.io/qt-6/qmultimap-obsolete.html#insertMulti

Use QMultiMap insert instead.

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
Upstream-Status: Submitted [https://github.com/sailfishos/sensorfw/pull/21]

 core/hybrisadaptor.cpp | 2 +-
 core/hybrisadaptor.h   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/hybrisadaptor.cpp b/core/hybrisadaptor.cpp
index 885f7b6..5128820 100644
--- a/core/hybrisadaptor.cpp
+++ b/core/hybrisadaptor.cpp
@@ -832,7 +832,7 @@ void HybrisManager::processSample(const sensors_event_t& data)
 void HybrisManager::registerAdaptor(HybrisAdaptor *adaptor)
 {
     if (!m_registeredAdaptors.values().contains(adaptor) && adaptor->isValid()) {
-        m_registeredAdaptors.insertMulti(adaptor->m_sensorType, adaptor);
+        m_registeredAdaptors.insert(adaptor->m_sensorType, adaptor);
     }
 }
 
diff --git a/core/hybrisadaptor.h b/core/hybrisadaptor.h
index 2b2af9e..0e5cfc7 100644
--- a/core/hybrisadaptor.h
+++ b/core/hybrisadaptor.h
@@ -186,7 +186,7 @@ public:
 private:
     // fields
     bool                          m_initialized;
-    QMap <int, HybrisAdaptor *>   m_registeredAdaptors; // type -> obj
+    QMultiMap <int, HybrisAdaptor *>   m_registeredAdaptors; // type -> obj
 
 #ifdef USE_BINDER
     // Binder backend
